File: Latihan/LatihanRawQuery/java/app/src/main/java/com/dicoding/picodiploma/mynoteapps/helper/SortUtils.java
package com.dicoding.picodiploma.mynoteapps.helper;

import androidx.sqlite.db.SimpleSQLiteQuery;

public class SortUtils {
    public static final String NEWEST = "Newest";
    public static final String OLDEST = "Oldest";
    public static final String RANDOM = "Random";

    public static SimpleSQLiteQuery getSortedQuery(String filter) {
        StringBuilder simpleQuery = new StringBuilder().append("SELECT * FROM note ");
        if (filter.equals(NEWEST)) {
            simpleQuery.append("ORDER BY id DESC");
        }
        else if (filter.equals(OLDEST)) {
            simpleQuery.append("ORDER BY id ASC");
        } else if (filter.equals(RANDOM)) {
            simpleQuery.append("ORDER BY RANDOM()");
        }
        return new SimpleSQLiteQuery(simpleQuery.toString());
    }
}

File: Latihan/LatihanRawQuery/kotlin/app/src/main/java/com/dicoding/picodiploma/mynoteapps/helper/SortUtils.kt
package com.dicoding.picodiploma.mynoteapps.helper

import androidx.sqlite.db.SimpleSQLiteQuery

object SortUtils {
    const val NEWEST = "Newest"
    const val OLDEST = "Oldest"
    const val RANDOM = "Random"

    fun getSortedQuery(filter: String): SimpleSQLiteQuery {
        val simpleQuery = StringBuilder().append("SELECT * FROM note ")
        if (filter == NEWEST) {
            simpleQuery.append("ORDER BY id DESC")
        } else if (filter == OLDEST) {
            simpleQuery.append("ORDER BY id ASC")
        } else if (filter == RANDOM) {
            simpleQuery.append("ORDER BY RANDOM()")
        }
        return SimpleSQLiteQuery(simpleQuery.toString())
    }
}